app/lib/frontend/backend.dart

```diff
@@ -297,6 +296,38 @@ class Backend {
       await analyzerClient.triggerAnalysis(package, latestVersion, <String>{});
     });
   }
+
+  /// Whether the current authenticated user (or [userId] if provided) is a
+  /// package admin (through direct uploaders list or publisher admin).
+  ///
+  /// Returns false if the user is not an admin.
+  Future<bool> isPackageAdmin(models.Package p, {String userId}) async {
+    userId ??= authenticatedUser?.userId;
```

Comment:
Do we use this default value else where but here, if not I suggest we make `userId` a required parameter.

Normally, I would prefer that when we read from `authenticatedUser`, if `authenticatedUser` is `null`, then we throw an `AuthenticationMissingException`.

Reply:
We do use `userId` unfortunately. I could do a second round of refactoring if I can get that out of the way, but first I'd rather have this merged, as it unblocks other publisher API implementation.

Reply:
Hm, actually, I don't think we can get rid of the `userId` parameter: when confirming the addition of an uploader, we do double-check if the originator of the invite is still and admin, and that needs the `userId` parameter.

Reply:
I suggest we make `userId` required, instead of optional. Such that we never reference `authenticatedUser` from within this function.

Normally, if `authenticatedUser` is `null` we should throw `AuthenticationMissingException`. That way we wouldn't make the mistake we made in `checkPackageAdmin` which will throw an authorization error instead of an authentication error if the user is not authenticated.